extensions/data-plane/data-plane-google-bigquery/src/main/java/org/eclipse/edc/connector/dataplane/gcp/bigquery/pipeline/BigQueryDataSource.java

```diff
@@ -0,0 +1,309 @@
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQuery.QueryResultsOption;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.JobId;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.edc.connector.dataplane.gcp.bigquery.params.BigQueryRequestParams;
+import org.eclipse.edc.connector.dataplane.spi.pipeline.DataSource;
+import org.eclipse.edc.connector.dataplane.spi.pipeline.StreamResult;
+import org.eclipse.edc.gcp.bigquery.BigQueryConfiguration;
+import org.eclipse.edc.gcp.bigquery.BigQueryPart;
+import org.eclipse.edc.gcp.bigquery.BigQueryTarget;
+import org.eclipse.edc.gcp.bigquery.service.BigQueryServiceSchema;
+import org.eclipse.edc.gcp.common.GcpException;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.stream.Stream;
+
+import static org.eclipse.edc.connector.dataplane.spi.pipeline.StreamResult.error;
+import static org.eclipse.edc.connector.dataplane.spi.pipeline.StreamResult.success;
+import static org.eclipse.edc.spi.constants.CoreConstants.EDC_NAMESPACE;
+
+/**
+ * Reads data using the BigQuery API (REST) and transfer it in JSON format.
+ */
+public class BigQueryDataSource implements DataSource {
+    static final int BIGQUERY_PAGE_SIZE = 4;
+    private BigQueryRequestParams params;
+    private String requestId;
+    private Monitor monitor;
+    private BigQueryConfiguration configuration;
+    private BigQueryTarget target;
+    private ExecutorService executorService;
+    private BigQuery bigQuery;
+    private GoogleCredentials credentials;
+
+    private BigQueryDataSource() {
+    }
+
+    @Override
+    public void close() {
+    }
+
+    @Override
+    public StreamResult<Stream<Part>> openPartStream() {
+        try {
+            var query = params.getQuery();
+            var sinkAddress = params.getSinkAddress();
+            var queryConfigBuilder = QueryJobConfiguration.newBuilder(query);
+            var destinationTable = params.getDestinationTable();
+            var customerName = sinkAddress.getStringProperty(BigQueryServiceSchema.CUSTOMER_NAME);
+            if (customerName != null) {
+                queryConfigBuilder.setLabels(ImmutableMap.of("customer", customerName));
```

Comment:
micro nit: use JDK classes, i.e. instead of `ImmutableMap.of` pls use `Map.of`. it's also immutable. We should try to limit the use of SDK classes wherever possible.

Reply:
Done